deltachat-ios/ViewModel/ChatListViewModel.swift

```diff
@@ -201,13 +201,18 @@ class ChatListViewModel: NSObject, ChatListViewModelProtocol {
         return nil
     }
 
-    func deleteChat(chatId: Int) -> Int {
+    func deleteChat(chatId: Int) -> Int? {
         // find index of chatId
         let indexToDelete = Array(0..<chatList.length).filter { chatList.getChatId(index: $0) == chatId }.first
+        let chat = dcContext.getChat(chatId: chatId)
         dcContext.deleteChat(chatId: chatId)
         updateChatList(notifyListener: false)
         safe_assert(indexToDelete != nil)
-        return indexToDelete ?? -1
+        // Do not return the index of the tableview cell for the selfTalk chat:
+        // Immediately after deleting the chat, a message to the device talk chat gets added.
+        // If the device talk chat was not existing it would be created and added to the table data source.
+        // That case invalidates the index.
+        return chat.isSelfTalk ? nil : indexToDelete
```

Comment:
hm, pretty special knowledge :)

i think, however, we're doing far too much here. [dc_delete_chat()](https://c.delta.chat/classdc__context__t.html#ad50eed96a11b113c4080886b2b748ff3) already emits the event DC_EVENT_MSGS_CHANGED that we observe and update the chatlist - neither the updateChatList() a few lines above is needed nor the index return value.